zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -1260,6 +1492,29 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
         $scope.paragraph.config.graph.scatter.xAxis = $scope.paragraph.result.columnNames[0];
       }
     }
+
+    /* try to find columns for the map logitude and latitude */
+    var i;
```

Comment:
There is some duplicate code here, it could be refactored like this:

```
var findDefaultMapCol = function(settingName, keyword) {
      var col;
      if (!$scope.paragraph.config.graph.map[settingName]) {
        for (var i = 0; i < $scope.paragraph.result.columnNames.length; ++i) {
          col = $scope.paragraph.result.columnNames[i];
          if (col.name.toUpperCase().indexOf(keyword) !== -1) {
            $scope.paragraph.config.graph.map[settingName] = col;
            break;
          }
        }
      }
    };

    findDefaultMapCol('lat', 'LAT');
    findDefaultMapCol('lng', 'LONG');
```